### ISSUE TITLE
Allow `max_lr` to be set per group

### ISSUE THREAD

**Issue by katsura-jp** [katsura-jp/pytorch-cosine-annealing-with-warmup#14]
From #11. 

Issue: if you have an optimizer with multiple groups with different learning rates, all of their learning rate values will get overridden by the `max_lr`.
